tests/phpunit/tests/option/wpLoadAlloptions.php

```diff
@@ -107,4 +107,22 @@ public function return_pre_cache_filter( $alloptions ) {
 		$this->alloptions = $alloptions;
 		return $this->alloptions;
 	}
+
+	/**
+	 * Test that you can shortcut the function with the filter "pre_wp_load_alloptions".
+	 *
+	 * @ticket 56045
+	 */
+	public function test_filter_pre_wp_load_alloptions_filter_is_called() {
+
+		add_filter( 'pre_wp_load_alloptions', array( $this, 'pre_wp_load_alloptions_filter' ) );
+
+		// Filter was called.
+		$this->assertEquals( array( 'filtered' => 'true' ), wp_load_alloptions() );
```

Comment:
```suggestion
		$this->assertSame( array( 'filtered' => 'true' ), wp_load_alloptions() );
```

Reply:
```suggestion
		$filter = new MockAction();

		add_filter( 'pre_wp_load_alloptions', array( &$filter, 'filter' ) );

		 wp_load_alloptions();

		$this->assertSame( 1, $filter->get_call_count() );
		$this->assertSame( array( 'pre_wp_load_alloptions' ), $filter->get_hook_names() );
```

@peterwilsoncc Taught me this trick a while back to test that a filter is indeed run.  I think it might be worth considering here for this case as well.  This would obviate the filter function as well.